Dockerfile

```diff
@@ -1,19 +1,34 @@
 # Build the SPIFFE CSI Driver binary
-FROM golang:1.19.3-alpine AS builder
+FROM --platform=${BUILDPLATFORM} golang:1.19.3-alpine AS base
 ARG GIT_TAG
 ARG GIT_COMMIT
 ARG GIT_DIRTY
 WORKDIR /code
 RUN apk --no-cache --update add make
 COPY go.* ./
-RUN go mod download
+RUN --mount=type=cache,target=/go/pkg/mod go mod download
 COPY . .
+
+# xx is a helper for cross-compilation
+# when bumping to a new version analyze the new version for security issues
+# then use crane to lookup the digest of that version so we are immutable
+# crane digest tonistiigi/xx:1.1.2
+FROM --platform=${BUILDPLATFORM} tonistiigi/xx@sha256:9dde7edeb9e4a957ce78be9f8c0fbabe0129bf5126933cd3574888f443731cda AS xx
```

Comment:
Since we have no CGO dependencies and thus can do very naive cross compilation with the standard go toolchain, is there a reason to use xx?

Reply:
It sets the correct variables like GO_ARCH and GO_PLATFORM based on the buildx arguments automatically.

Reply:
Fair enough.